src/app/node-data/dialog/component.ts

```diff
@@ -163,6 +163,7 @@ export class NodeDataDialogComponent extends BaseFormValidator implements OnInit
   }
 
   private _isRecreationWarningVisible(): boolean {
-    return this.mode === Mode.Edit && !_.isEqual(objectDiff(this._data.initialNodeData, this._output.nodeData), {});
+    const diff = objectDiff(this._data.initialNodeData, this._output.nodeData);
+    return this.mode === Mode.Edit && !_.isEqual(diff, {}) && !(Object.keys(diff).length === 1 && !!diff.count);
```

Comment:
This looks complex. What is the expected result here? If we want to show the warning only if the replica number has changed then why we cannot use `!==` instead of `objectDiff()`?

Reply:
Let's also keep in mind that this was requested in some issue and at the moment we have an issue to fix bug in it: https://github.com/kubermatic/dashboard/issues/2496.

Reply:
we **don't** want to show this warning when replica number has changed. But for all other changes we want to display it

Reply:
Ah, okay. You're right. I'd leave some comment as it's not really easy to understand what is going on here.

Reply:
Sure, makes sense. Added it.

Reply:
I also have some reservations about using `!!` on a number property. If the `this._data.initialNodeData.count` would be 0 (it seems to be defaulted at the moment and it works fine) then we may get invalid result. I'd rather check against `undefined` or even if the `Object.keys` contains the `count` property. What do you think?

Reply:
You mean something like this?

```
const diff = objectDiff(this._data.initialNodeData, this._output.nodeData);
return this.mode === Mode.Edit && !_.isEqual(diff, {}) && 
       !(Object.keys(diff).length === 1 && !!diff.hasOwnProperty('count'));
```

Reply:
Exactly, feels a bit safer :) BTW, we don't need `!!` on `hasOwnProperty` as it returns boolean anyways.

Reply:
Yes, you're right.
Commited it and only changed it a bit because of `Do not access Object.prototype method 'hasOwnProperty' from target object  - no-prototype-builtins` error ;)

Reply:
Good, let's merge.